crates/ruff_cli/src/diagnostics.rs

```diff
@@ -124,33 +137,18 @@ impl AddAssign for Diagnostics {
 /// Read a Jupyter Notebook from disk.
 ///
 /// Returns either an indexed Python Jupyter Notebook or a diagnostic (which is empty if we skip).
-fn notebook_from_path(path: &Path) -> Result<Notebook, Box<Diagnostics>> {
-    let notebook = match Notebook::from_path(path) {
-        Ok(notebook) => {
-            if !notebook.is_python_notebook() {
-                // Not a python notebook, this could e.g. be an R notebook which we want to just skip.
-                debug!(
-                    "Skipping {} because it's not a Python notebook",
-                    path.display()
-                );
-                return Err(Box::default());
-            }
-            notebook
-        }
-        Err(diagnostic) => {
-            // Failed to read the jupyter notebook
-            return Err(Box::new(Diagnostics {
-                messages: vec![Message::from_diagnostic(
-                    *diagnostic,
-                    SourceFileBuilder::new(path.to_string_lossy().as_ref(), "").finish(),
-                    TextSize::default(),
-                )],
-                ..Diagnostics::default()
-            }));
-        }
-    };
-
-    Ok(notebook)
+fn notebook_from_path(path: &Path) -> Result<Option<Notebook>, NotebookError> {
```

Comment:
Return `Result<Option<Notebook>>` instead of using empty `Diagnostics` to indicate the "we parsed the notebook, but it's not a Python notebook, so it should be skipped" case.